### PR TITLE
Export assert_fail with FMT_API. This fixes dll build.

### PR DESCRIPTION
**Pull request by egorpugin**
See https://github.com/fmtlib/fmt/issues/1445

I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

File: include/fmt/format-inl.h [hunk 1]
```diff
@@ -55,7 +55,7 @@ inline fmt::internal::null<> strerror_s(char*, std::size_t, ...) { return {}; }
 FMT_BEGIN_NAMESPACE
 namespace internal {
 
-FMT_FUNC void assert_fail(const char* file, int line, const char* message) {
+FMT_API FMT_FUNC void assert_fail(const char* file, int line, const char* message) {
```
Comment (vitaut) [inline include/fmt/format-inl.h]: I think `FMT_API` is not needed here.

**Comment by vitaut** [comment]
Thanks!

File: include/fmt/format-inl.h (hunk 1 above)
Comment (egorpugin) [reply include/fmt/format-inl.h]: Oh, maybe. First, I thought header goes to users, but it seems it is fmt internal stuff?

Comment (egorpugin) [reply include/fmt/format-inl.h]: Yep, tested, works without it.

Comment (vitaut) [reply include/fmt/format-inl.h]: It's for the opt-in header-only mode and effectively an internal header.

**Pull request merged**